### PR TITLE
Sitemap generation

### PR DESCRIPTION
**Pull request by leftclickben**
Generates the sitemap and puts it in a local file.  Next PR will push the file to S3.  Then there will be packaging for execution in Lambda.

### CODE REVIEW

File: .jshintrc [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  "notypeof": true,
+  "singleGroups": false,
+  "undef": true,
+
+  // environment
+  "debug": true,
+  "devel": true,
+
+  // globals
+  "browser": true,
```
Comment (kehh) [inline .jshintrc]: really?

File: bin/generate.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        request(
+            config.service.protocol + '://' + config.service.username + ':' + config.service.password + '@' + config.service.hostname + '/service.php/simple/sitemap_search?q=*&noCache=1',
+            function (err, response) {
+                var data;
+
+                if (err) {
+                    return callback(err);
+                }
+
+                data = JSON.parse(response.body);
```
Comment (kehh) [inline bin/generate.js:32]: If the response isn't JSON won't this vomit here? (although it would presumably vomit in a predictable and easy to way)

**Comment by kehh** [comment]
Just two questions.

File: .jshintrc (hunk 1 above)
Comment (leftclickben) [inline .jshintrc]: Copypasta error, will fix.

File: bin/generate.js (hunk 2 above)
Comment (leftclickben) [inline bin/generate.js:32]: I think it should vomit here if the web service returns something that isn't JSON.

**Pull request merged**